build/wd_compile_cache.bzl

```diff
@@ -0,0 +1,57 @@
+"""
+Bazel rules for generating compilation caches from a list of input files.
+
+This module provides rules and macros for creating compilation caches, which can be used
+to improve build performance by caching compilation artifacts.
+"""
+
+load("@bazel_skylib//rules:run_binary.bzl", "run_binary")
+
+def _gen_compile_cache_impl(ctx):
+    """Implementation of the gen_compile_cache rule.
+
+    This function:
+    1. Creates a file containing the list of input files
+    2. Runs a binary to generate the compilation cache
+    """
+
+    file_list_path = "{}-compile-cache-file-list".format(ctx.attr.name)
+    compile_cache_output = "{}-compile-cache".format(ctx.attr.name)
+    ctx.actions.write(
+        output = file_list_path,
+        content = "\n".join(ctx.attr.data) + "\n",
+    )
+    run_binary(
+        ctx.attr.name,
+        srcs = [],
+        outs = [compile_cache_output],
+        args = [file_list_path],
+        tool = "create_compile_cache_bin",
+    )
+    return compile_cache_output
+
+gen_compile_cache = rule(
+    implementation = _gen_compile_cache_impl,
+    attrs = {
+        "name": attr.string(mandatory = True, doc = "name"),
+        "data": attr.string_list(mandatory = True, doc = "files to create compile cache for"),
+    },
+    target_compatible_with = select({
+        "@platforms//os:linux": [],
+        "//conditions:default": ["@platforms//:incompatible"],
```

Comment:
sorry, is there any real reason we avoid mac+windows here? I don't see why wouldn't this work?

Reply:
my initial implementation didn't use KJ_MAIN and I wanted to avoid Windows specific code because of that. since the implementation is now different, I guess there is no reason to avoid mac+windows.